test/unit/cookie.test.js

```diff
@@ -46,6 +46,34 @@ describe('Cookie', function () {
             expect(strCookie.valueOf()).to.equal('%E0%A4%A');
         });
 
+        it('should not set name value that are not of string type', function () {
+            var strCookie = new Cookie({
+                name: 1,
+                value: 'bar'
+            });
+
+            expect(strCookie).to.deep.include({
+                value: 'bar'
+            });
+            expect(strCookie.valueOf()).to.equal('bar');
+        });
+
+        it('should use key value as name in case of cookie name not specified', function () {
+            var testCookie = new Cookie({
+                    key: 'testCookie',
+                    expires: 1502442248,
+                    hostOnly: false,
+                    httpOnly: false,
+                    path: '/',
+                    secure: false,
+                    session: false,
+                    value: 'fooTest'
+                }),
+                unparsedSingle = Cookie.unparseSingle(testCookie);
+
+            expect(unparsedSingle).to.equal(testCookie.name + '=' + testCookie.value);
```

Comment:
Assert the exact string value instead of concatenating.

Reply:
Fixed.